### DISCUSSION TITLE
Wait for page load completes - How to achieve this.

### DISCUSSION THREAD

**Discussion by Jaspher16** [getgauge/taiko#2661] [General]
My application lot of navigations and it applies too much timeouts when ever I perform navigation or clicks.

Is there any method I can use to confirm the page navigates or done with loading?

**Comment by NivedhaSenthil** [comment]
Taiko by default waits for page load to complete by waiting for `loadEventFired ` event for navigation apis like goto and reload . If any other action leads to page navigation you could explicitly wait for it as mentioned here,
https://docs.taiko.dev/frequently_asked_questions/#how-can-i-customize-taiko's-wait-events%3F
